afterpay/src/main/kotlin/com/afterpay/android/Afterpay.kt

```diff
@@ -258,12 +268,127 @@ object Afterpay {
+        }
+
+        // requesting checkout data failed
+        checkoutResponseResult.onFailure { error: Throwable ->
+            return failure(error)
+        }
+
+        // requesting checkout data success: sign the checkout response token
+        checkoutResponseResult.onSuccess { result: CheckoutV3.Response ->
+            AfterpayCashAppCheckout.performSignPaymentRequest(result.token)
+                .let { cashAppSignOrderResult: CashAppSignOrderResult ->
+                    return when (cashAppSignOrderResult) {
+                        // signing failed
+                        is Failure -> failure(cashAppSignOrderResult.error)
+
+                        // signing success
+                        is Success -> {
+                            // map Result<CashAppSignOrderResult> to Result<CheckoutV3CashAppPay>
+                            success(
+                                CheckoutV3CashAppPay(
+                                    token = result.token,
+                                    singleUseCardToken = result.singleUseCardToken,
+                                    amount = cashAppSignOrderResult.response.amount,
+                                    redirectUri = cashAppSignOrderResult.response.redirectUri,
+                                    merchantId = cashAppSignOrderResult.response.merchantId,
+                                    brandId = cashAppSignOrderResult.response.brandId,
+                                    jwt = cashAppSignOrderResult.response.jwt,
+                                ),
+                            )
+                        }
+                    }
+                }
+        }
+
+        // should never happen, compiler doesn't know success and failure are only options
+        throw IllegalStateException()
+    }
+
+    /**
+     * Confirm that checkout was completed with Cash App Pay SDK
+     */
+    suspend fun confirmCheckoutV3WithCashAppPay(
+        customerId: String,
+        grantId: String,
+        token: String,
+        singleUseCardToken: String,
+        jwt: String,
+        configuration: CheckoutV3Configuration? = checkoutV3Configuration,
+    ): Result<CheckoutV3Data> {
+        return runCatching {
+            requireNotNull(configuration) {
+                "`configuration` must be set via `setCheckoutV3Configuration` or passed into this function"
+            }
+
+            val confirmUrl = configuration.v3CheckoutConfirmationUrl
+
+            val request = CashAppPayRequest(
+                token = token,
+                singleUseCardToken = singleUseCardToken,
+                cashAppPspInfo = CashAppPayRequest.CashAppPspInfo(
+                    externalCustomerId = customerId,
+                    externalGrantId = grantId,
+                    jwt = jwt,
+                ),
+            )
+
+            val response = withContext(Dispatchers.IO) {
+                ApiV3.request<CashAppPayResponse, CashAppPayRequest>(
+                    confirmUrl,
```

Comment:
nit: let's add named parameters here too, for consistency with the rest of the newer code